lib/pure/times.nim

```diff
@@ -1288,32 +1289,20 @@ proc timeToTimeInfo*(t: Time): TimeInfo =
 
 proc timeToTimeInterval*(t: Time): TimeInterval =
   ## Converts a Time to a TimeInterval.
+  # Milliseconds not available from Time
+  var tInfo = t.getLocalTime()
+  initInterval(0, tInfo.second, tInfo.minute, tInfo.hour, tInfo.weekday.ord, tInfo.month.ord, tInfo.year)
+  
+proc initTime*(s: string, layout: string = "yyyy-MM-dd hh:mm:ss"): Time =
+  ## Converts a string to Time.
+  ## Default layout is ISO 8601
+  result = s.parse(layout).timeInfoToTime
 
-  let
-    secs = t.toSeconds().int
-    daysSinceEpoch = secs div secondsInDay
-    (yearsSinceEpoch, daysRemaining) = countYearsAndDays(daysSinceEpoch)
-    daySeconds = secs mod secondsInDay
-
-  result.years = yearsSinceEpoch + epochStartYear
-
-  var
-    mon = mJan
-    days = daysRemaining
-    daysInMonth = getDaysInMonth(mon, result.years)
-
-  # calculate month and day remainder
-  while days > daysInMonth and mon <= mDec:
-    days -= daysInMonth
-    mon.inc
-    daysInMonth = getDaysInMonth(mon, result.years)
+proc initTimeFromStr*(s: string): Time =
```

Comment:
Aren't both procs creating a ``Time`` out of a string? You can likely merge together. These procedures are just a shortcut though and I'm not sure they are worth it.